pkg/monitoring/metrics.go

```diff
@@ -0,0 +1,87 @@
+package monitoring
+
+import (
+	"strings"
+
+	api "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type PodStatus string
+
+const (
+	PodStatusInitializing    PodStatus = "Initializing"
+	PodStatusReady           PodStatus = "Ready"
+	PodStatusPending         PodStatus = "Pending"
+	PodStatusError           PodStatus = "Error"
+	PodStatusDecommissioning PodStatus = "Decommissioning"
+)
+
+func getPodStatus(pod *corev1.Pod) PodStatus {
+	status := PodStatusReady
+
+	if pod.Labels[api.CassNodeState] == "Decommissioning" {
+		return PodStatusDecommissioning
+	}
+
+	switch pod.Status.Phase {
+	case corev1.PodPending:
+		return PodStatusPending
+	case corev1.PodFailed:
+		return PodStatusError
+	case corev1.PodRunning:
+	default:
+	}
+
+	allContainersReady := true
+
+	for _, s := range pod.Status.ContainerStatuses {
+		if !s.Ready {
+			allContainersReady = false
+			break
+		}
+	}
+
+	if !allContainersReady {
+		return PodStatusInitializing
+	}
+
+	return status
+}
+
+var (
+	PodStatusVec *prometheus.GaugeVec
+)
+
+func init() {
+	podVec := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "cass_operator",
```

Comment:
**Issue**: cass-operator is hardcoded here but we will have DCs in different namespaces.
Even if this `Namespace` notion is not equivalent to the k8s namespace, we lack a label in the metrics to filter namespaces. We could two dcs with the same cluster and dc names but in different namespaces and I don't see how we could differentiate them.

Reply:
This "namespace" is actually just part of the metric name, the structure is ``Namespace_Subsystem_Name``. We can add namespace as a label or we can of course keep changing the metric name also, but I would assume label is easier to query and index in any Prometheus compatible system instead of changing the prefix of a metric name.